src/node/src/common.js

```diff
@@ -42,39 +42,73 @@
 var _ = require('lodash');
 
 /**
- * Get a function that deserializes a specific type of protobuf.
- * @param {function()} cls The constructor of the message type to deserialize
- * @param {bool=} binaryAsBase64 Deserialize bytes fields as base64 strings
- *     instead of Buffers. Defaults to false
+ * Build the options object to pass to ProtoBuf.Message.asJSON
+ * @param {bool=} enumsAsStrings Deserialize enum fields as key strings
+ *     instead of numbers. Defaults to false
  * @param {bool=} longsAsStrings Deserialize long values as strings instead of
  *     objects. Defaults to true
- * @return {function(Buffer):cls} The deserialization function
+ * @param {bool=} binaryAsBase64 Deserialize bytes values as base64 strings
+ *     instead of Buffers. Defaults to false
+ * @return {Object} The options to pass to ProtoBuf.Message.asJSON
  */
-exports.deserializeCls = function deserializeCls(cls, binaryAsBase64,
-                                                 longsAsStrings) {
-  if (binaryAsBase64 === undefined || binaryAsBase64 === null) {
-    binaryAsBase64 = false;
-  }
+exports.buildAsJSONOptions = function buildAsJSONOptions(enumsAsStrings,
+                                                         longsAsStrings,
+                                                         binaryAsBase64) {
+  enumsAsStrings = !!enumsAsStrings;
```

Comment:
If `enumsAsStrings` was the default in Protobuf.js 5, it should also be the default here.